dubbo-registry/dubbo-registry-consul/src/main/java/org/apache/dubbo/registry/consul/ConsulRegistry.java

```diff
@@ -154,6 +156,24 @@ public void doUnsubscribe(URL url, NotifyListener listener) {
         notifier.stop();
     }
 
+    @Override
+    public List<URL> lookup(URL url) {
+        if (url == null) {
+            throw new IllegalArgumentException("lookup url == null");
+        }
+        try {
+            String service = url.getServiceKey();
+            Response<List<HealthService>> result = client.getHealthServices(service, HealthServicesRequest.newBuilder().setTag(SERVICE_TAG).build());
```

Comment:
is it possible to reuse **getHealthServices(java.lang.String, long, int)**?

Reply:
It seems the tcp check never work correctly in consulRegistry. Is there a must that we use tcp check, instead of a http ttl check? I used ttl check, with onlyPass=true, then things went correct.

Reply:
with this tcp Check  configuration,  client.getHealthService shows the check is critical. What's your opinion sir?

Reply:
tcp check does work on my side. Using TCP check instead of others does have performance consideration in it.

Reply:
one more question, why this change is related to TCP check (I guess you mean health check by TCP, am I right?)

Reply:
yes, consul health check by TCP.  after we register consul service with tcp check, we cannot get  a passing=true HealthService response every time. the ConsulRegistryTest.class now does not involve this logic actually.

Reply:
I will spend more time on this :)

Reply:
could u pls check this code(which is implement by me):  https://github.com/baidu/brpc-java/blob/97e6d71d8b83b061384afa9d139a2dc5da841451/brpc-java-naming-consul/src/main/java/com/baidu/brpc/naming/consul/ConsulNamingService.java#L126

I believe we need to explicitly healthCheckPass the Check(by checkId) either. Then we can correctly getHealthServices(java.lang.String, long, int) by params: onlyPassing = true

Reply:
my first version does use **agentCheckPass** instead of **setTcp**.  Not sure why TCP check doesn't work perfectly on your side, would you mind share me your demo app so that I could reproduce. 

If you believe **agentCheckPass** is more robust, would you mind add it back? You could refer my original change: https://github.com/beiwei30/incubator-dubbo/commit/2f828965339176ef76ddb3b91fb74c90d3f3f051 and enhance it. Luckily I didn't delete it from my repo :)

Reply:
@Moriadry should I merge this pull request first? then you can submit another pull request to follow up with https://github.com/beiwei30/incubator-dubbo/commit/2f828965339176ef76ddb3b91fb74c90d3f3f051. What do you think?

Reply:
ok, you can merge this, another pr will be provided soon

Reply:
sure. I will merge it. Thanks for following up. I will add a new samples for consul to test consul TCP check, will ping you once it's ready.